source/lib/calendar-helper.ts

```diff
@@ -14,3 +14,7 @@ export function getUrl(id: number, userconfig: Userconfig): string {
 export function getUrlFromContext(ctx: MyContext): string {
 	return getUrl(ctx.from!.id, ctx.userconfig.mine);
 }
+
+export function getUserEventIdsFromContext(ctx: MyContext): EventId[] {
+	return Object.keys(ctx.userconfig.mine.events) as EventId[];
+}
```

Comment:
Generische Form wäre das hier:

```ts
export function typedKeys<T>(obj: T): Array<keyof T> {
  return Object.keys(obj) as Array<keyof T>
}
```

Magst du eine `source/lib/(javascript-)helper.ts` anlegen mit der Funktion? Not sure, ob mit oder ohne javascript- im Namen, vielleicht mit ums klarer zu machen, wofür helper… Gibt ja schon ein paar.
Hab ich in ein zwei mehr Projekten für genau den Zweck, generell überall im Code hilfreich und resultiert aus Dingen, die JavaScript nicht gut genug kann, TypeScript aber potenziell besser.

Reply:
Also doch wieder die zusätzliche Funktion?

Reply:
Ne, ich finds mit `typesKeys(ctx.userconfig.mine.events)` lesbarer, weil ich auf einen Blick weiß, dass das Object.keys(…) ist und nicht erst drüber nachdenken muss, was `getUserEventIdsFromContext` bedeuten könnte. Dadurch das es häufiger genutzt wurde als ich dachte, ist es an mehr stellen lesbarer als ich erst gedacht hätte.